### ISSUE TITLE
Docs: `aria-disabled` does not disable button.

### ISSUE THREAD

**Issue by byseop** [vercel/next.js#63887]
### What is the improvement or update you wish to see?

I would like the examples on [this page](https://nextjs.org/docs/app/building-your-application/authentication) to be corrected.

```tsx
<button aria-disabled={pending} type="submit">
  Login
</button>
```
The `aria-disabled` attribute on the button tag does not actually make the button tag disabled.

### Is there any context that might help us understand?

The `aria-disabled` attribute of the button tag should be modified to `disabled`.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/building-your-application/authentication

**Comment by byseop** [comment]
I fixed it and PR. #63888

**Comment by jenewland1999** [comment]
While `aria-disabled` does indeed not prevent you from clicking the button itself, it's important to understand why `aria-disabled` is used to begin with. There's a great article on CSS Tricks about it that's worth a read https://css-tricks.com/making-disabled-buttons-more-inclusive/.

**Comment by byseop** [comment]
> While `aria-disabled` does indeed not prevent you from clicking the button itself, it's important to understand why `aria-disabled` is used to begin with. There's a great article on CSS Tricks about it that's worth a read https://css-tricks.com/making-disabled-buttons-more-inclusive/.

@jenewland1999 Thank you for the additional information. 

In my opinion, `disabled` seems more appropriate than `aria-disabled` for common examples.

**Comment by jenewland1999** [comment]
> > While `aria-disabled` does indeed not prevent you from clicking the button itself, it's important to understand why `aria-disabled` is used to begin with. There's a great article on CSS Tricks about it that's worth a read https://css-tricks.com/making-disabled-buttons-more-inclusive/.
> 
> @jenewland1999 Thank you for the additional information. 
> 
> In my opinion, `disabled` seems more appropriate than `aria-disabled` for common examples.
> 
> The `aria-***` attributes will inform people using assistive technologies, such as screen readers, that such elements are not meant to be editable or otherwise operable.
> 
> 
> [MDN article explains](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled)
> > Unlike HTML's [disabled](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#disabled) Boolean attribute, which will communicate a form control as semantically being disabled, change its styling to reflect its state and suppress all functionality along with disallowing the element's value from participating in form submission, the aria-disabled="true" only semantically exposes these elements as being disabled. Web developers must manually ensure such elements have their functionality suppressed when exposed to the disabled state.

I don't disagree that the `disabled` attribute is more common/familiar and likely easier for the examples because it gives intended behaviour. That said the follow up comments on PR #63852 suggest the intention might be purposefully to promote more accessible examples/docs. 

That said it is a bit of a grey area.

**Comment by byseop** [comment]
> > > While `aria-disabled` does indeed not prevent you from clicking the button itself, it's important to understand why `aria-disabled` is used to begin with. There's a great article on CSS Tricks about it that's worth a read https://css-tricks.com/making-disabled-buttons-more-inclusive/.
> > 
> > @jenewland1999 Thank you for the additional information. 
> > 
> > In my opinion, `disabled` seems more appropriate than `aria-disabled` for common examples.
> > 
> > The `aria-***` attributes will inform people using assistive technologies, such as screen readers, that such elements are not meant to be editable or otherwise operable.
> > 
> > 
> > [MDN article explains](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled)
> > > Unlike HTML's [disabled](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#disabled) Boolean attribute, which will communicate a form control as semantically being disabled, change its styling to reflect its state and suppress all functionality along with disallowing the element's value from participating in form submission, the aria-disabled="true" only semantically exposes these elements as being disabled. Web developers must manually ensure such elements have their functionality suppressed when exposed to the disabled state.
> 
> I don't disagree that the `disabled` attribute is more common/familiar and likely easier for the examples because it gives intended behaviour. That said the follow up comments on PR #63852 suggest the intention might be purposefully to promote more accessible examples/docs. 
> 
> That said it is a bit of a grey area.

The suggestion in that pr also looks good. I will change and update soon
